src/graph.rs

```diff
@@ -1515,7 +1521,12 @@ impl<'a> Builder<'a> {
         .graph
         .module_slots
         .insert(specifier.clone(), ModuleSlot::Pending);
-      self.pending.push(self.loader.load(specifier, is_dynamic));
+      let specifier = specifier.clone();
+      let fut = self
+        .loader
+        .load(&specifier, is_dynamic)
+        .map(move |res| (specifier, res));
+      self.pending.push(Box::pin(fut));
```

Comment:
@kitsonk Please see here. The specifier is still added to the future stored in the `FuturesUnordered`. It is just provided by the inserter (`deno_graph`), instead of the implementer of `Loader`.

Reply:
This is what my comment about the extra heap allocation was about :-)

Reply:
Well, if I actually read the PR versus just the description! 😞  🤦  This is what I get for reading things on my phone late at night.